### PR TITLE
Enable all supported IAST tests for Java

### PR DESCRIPTION
**Pull request by smola**
## Description

*Please include a short summary of the change*

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

File: tests/appsec/iast/source/test_iast_request_header.py [hunk 1]
```diff
@@ -30,6 +34,7 @@ class TestRequestHeader:
     def setup_header(self):
         self.r = weblog.get("/iast/source/header/test", headers={"random-key": "header-source"})
 
+    @bug(context.weblog_variant == "jersey-grizzly2", reason="name field of source not set")
```
Comment (GustavoCaso) [inline tests/appsec/iast/source/test_iast_request_header.py:37]: Should we also add the library java condition? I know the weblog variant is unique for java, but adding that extra context would help readers know which library has the bug

File: tests/appsec/iast/source/test_iast_request_parameter.py [hunk 2]
```diff
@@ -29,6 +33,7 @@ class TestRequestParameter:
     def setup_parameter(self):
         self.r = weblog.post("/iast/source/parameter/test", data={"source": "parameter"})
 
+    @bug(context.weblog_variant == "jersey-grizzly2", reason="name field of source not set")
```
Comment (GustavoCaso) [inline tests/appsec/iast/source/test_iast_request_parameter.py:36]: Same as https://github.com/DataDog/system-tests/pull/1079/files#r1169603252

File: tests/appsec/iast/source/test_iast_request_header.py (hunk 1 above)
Comment (cbeauchesne) [reply tests/appsec/iast/source/test_iast_request_header.py:37]: Weblog names are globally unique, but you're right, nothing enforce this, apart of my vigilance. so it won't hurt :)

**Comment by cbeauchesne** [comment]
Waiting #1081 to be merged before approving this one.

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**